src/services/githubauth.service.js

```diff
@@ -124,6 +125,44 @@ export default class GithubAuthenticationService {
     return tokenCreated;
   }
 
+  /**
+   * First exchanges code and state for a github token.
+   * Then adds a github token for the given user into the githubTokens table. If a token already
+   * exists for the user, it will automatically update the old one.
+   * @param userId the user id to add the token for
+   * @param code the code to exchange for token
+   * @param state the state to verify identity
+   * @param username the github username of the user
+   * @returns {Object} the token that was created
+   */
+  async getAndSetGithubTokenForUser(userId, code, state, username) {
+    try {
+      const secrets = config.get('oauth_secrets');
+      const res = await fetch('https://cors-anywhere.herokuapp.com/https://github.com/login/oauth/access_token', {
+        method: 'POST',
+        body: JSON.stringify({
+          client_id: '6b57706945fa18ee0397',
+          client_secret: secrets.github,
+          code,
+          state
+        }),
+        headers: new fetch.Headers({
+          'Content-Type': 'application/json',
+          origin: 'http://localhost:8080/',
```

Comment:
can we move this url to the config file? That way we don't have to modify source code when we move to production.

Reply:
the origin is being passed to satisfy the cor-anywhere which I expect to be changed and if not the value doesn't matter as far as I can tell

Reply:
sounds good

Reply:
If that's the case, at least add a comment to that effect. Someone maintaining this code later may scratch their head wondering why the decision was made. On the other hand, it wouldn't be hard to move this to a config file...